hazelcast/src/main/java/com/hazelcast/config/security/SimpleAuthenticationConfig.java

```diff
@@ -0,0 +1,274 @@
+    }
+
+    /**
+     * Returns role names assigned to the given user.
+     *
+     * @param username username
+     * @return configured roles or null if the user doesn't exist
+     */
+    public Set<String> getRoles(@Nonnull String username) {
+        UserDto user = userMap.get(requireNonNull(username, "Username has to be provided"));
+        return user == null ? null : new HashSet<>(user.roles);
+    }
+
+    /**
+     * Replaces the users with ones from the given map.
+     *
+     * @param map map of new users
+     * @return this object
+     */
+    public SimpleAuthenticationConfig setUserMap(@Nullable Map<String, UserDto> map) {
+        userMap.clear();
+        if (map != null) {
+            userMap.putAll(map);
+        }
+        return self();
+    }
+
+    @Override
+    protected Properties initLoginModuleProperties() {
+        Properties props = super.initLoginModuleProperties();
+        setIfConfigured(props, "roleSeparator", roleSeparator);
+        String rs = roleSeparator != null ? roleSeparator : ",";
+        for (Map.Entry<String, UserDto> entry : userMap.entrySet()) {
+            String name = entry.getKey();
+            UserDto user = entry.getValue();
+            setIfConfigured(props, "password." + name, user.password);
+            setIfConfigured(props, "roles." + name, user.roles.stream().collect(joining(rs)));
+        }
+        return props;
+    }
+
+    @Override
+    public LoginModuleConfig[] asLoginModuleConfigs() {
+        LoginModuleConfig loginModuleConfig = new LoginModuleConfig(
+                "com.hazelcast.security.loginimpl.SimplePropertiesLoginModule", LoginModuleUsage.REQUIRED);
+
+        loginModuleConfig.setProperties(initLoginModuleProperties());
+
+        return new LoginModuleConfig[] { loginModuleConfig };
+    }
+
+    @Override
+    public String toString() {
+        return "SimpleAuthenticationConfig [userMap=" + userMap + ", roleSeparator=" + roleSeparator + "]";
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = super.hashCode();
+        result = prime * result + Objects.hash(roleSeparator, userMap);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!super.equals(obj)) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        SimpleAuthenticationConfig other = (SimpleAuthenticationConfig) obj;
+        return Objects.equals(roleSeparator, other.roleSeparator) && Objects.equals(userMap, other.userMap);
+    }
+
+    @Override
+    protected SimpleAuthenticationConfig self() {
+        return this;
+    }
+
+    /**
+     * Helper object representing user attributes (i.e. password and roles) in the {@link SimpleAuthenticationConfig}.
+     */
+    public static class UserDto {
```

Comment:
I would use this class as an immutable instance, just saves the password/roles in the constructor and does not offer any update mechanisms.

Reply:
:+1: updated